presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -283,41 +354,51 @@ public void testInnerJoinWithSpill(boolean probeHashEnabled, List<WhenSpill> whe
                 runDriver(executor, buildDriver);
             }
 
-            // process join
-            Iterator<Page> input = probePages.build().iterator();
-            List<Page> actualPages = new ArrayList<>();
-            actualPages.addAll(toPagesPartial(joinOperator, singletonIterator(input.next())));
+            ValuesOperatorFactory valuesOperatorFactory = new ValuesOperatorFactory(17, new PlanNodeId("values"), probePages.getTypes(), probePages.build());
+
+            PageBuffer pageBuffer = new PageBuffer(10);
+            PageBufferOperatorFactory pageBufferOperatorFactory = new PageBufferOperatorFactory(18, new PlanNodeId("pageBuffer"), pageBuffer);
+
+            Driver ljoDriver = new Driver(ljoDriverContext,
+                                          valuesOperatorFactory.createOperator(ljoDriverContext),
+                                          joinOperator,
+                                          pageBufferOperatorFactory.createOperator(ljoDriverContext));
+
+            while (!called.get()) { // process first row of first page of LookupJoinOperator
+                ljoDriver.getDriverContext().getYieldSignal().setWithDelay(10000, ljoDriver.getDriverContext().getYieldExecutor());
+                tryGetFutureValue(ljoDriver.process(), 10, TimeUnit.MILLISECONDS);
```

Comment:
you more or less ignore the returned future, so what about replacing this line with just `ljoDriver.process();` (as is common elsewhere in this file) ?